test/gtest/ucs/test_type.cc

```diff
@@ -62,23 +62,23 @@ UCS_TEST_F(test_type, serialize) {
     /* Pack */
     uint64_t *p64;
     void *pack_ptr = &data[0];
-    ucs_serialize_put(pack_ptr, uint8_t, values[0]);
-    ucs_serialize_put(pack_ptr, uint32_t, values[1]);
-    ucs_serialize_put(pack_ptr, uint64_t, values[2]);
-    ucs_serialize_next_typed_ptr(pack_ptr, p64);
+    ucs_serialize_put(&pack_ptr, uint8_t, values[0]);
+    ucs_serialize_put(&pack_ptr, uint32_t, values[1]);
+    ucs_serialize_put(&pack_ptr, uint64_t, values[2]);
+    ucs_serialize_next_typed_ptr(&pack_ptr, p64);
     *p64 = values[3];
-    ucs_serialize_put(pack_ptr, uint64_t, values[4]);
+    ucs_serialize_put(&pack_ptr, uint64_t, values[4]);
```

Comment:
> WDYT about latest version?

I don't like this with auto advancing `pack_ptr`, mixing `ucs_serialize_put` and `ucs_serialize_next_[typed_]ptr` seems not clear and counter intuitive to me

Reply:
what if we remove ucs_serialize_put/get, keep only ucs_serialize_next_typed_ptr (maybe rename it), and use code will do like this:
```
*ucs_serialize_next_typed_ptr(&pack_ptr, uint64_t) = values[2];
```
or
```
*ucs_serialize_next(&pack_ptr, uint64_t) = values[2];
```

Reply:
this sounds simpler